src/ui/public/parse_query/parse_query.js

```diff
@@ -1,27 +1,51 @@
-import { toUser } from 'ui/parse_query/lib/to_user';
-import { ParseQueryLibFromUserProvider } from 'ui/parse_query/lib/from_user';
-
-import { uiModules } from 'ui/modules';
-uiModules
-  .get('kibana')
-  .directive('parseQuery', function (Private) {
-    const fromUser = Private(ParseQueryLibFromUserProvider);
-
-    return {
-      restrict: 'A',
-      require: 'ngModel',
-      scope: {
-        'ngModel': '='
-      },
-      link: function ($scope, elem, attr, ngModel) {
-        const init = function () {
-          $scope.ngModel = fromUser($scope.ngModel);
-        };
-
-        ngModel.$parsers.push(fromUser);
-        ngModel.$formatters.push(toUser);
-
-        init();
-      }
-    };
-  });
+import _ from 'lodash';
+import angular from 'angular';
+
+export function getDefaultQuery() {
+  return { match_all: {} };
+}
+
+export function isDefaultQuery(query) {
+  return _.isEqual(query, getDefaultQuery());
+}
+
+export function getTextQuery(query) {
+  return {
+    query_string: { query }
+  };
+}
+
+export function isTextQuery(query) {
+  return _.has(query, 'query_string');
+}
+
+export function getQueryText(query) {
+  return _.get(query, ['query_string', 'query']) || '';
+}
+
+export function parseQuery(query) {
+  if (!_.isString(query) || query.trim() === '') {
+    return getDefaultQuery();
```

Comment:
Yeah, the only place I could tell that this would ever happen was in the [init of the parse query directive](https://github.com/elastic/kibana/blob/7a918f17d2da95a63b20c05dcf629cc5b0567b07/src/ui/public/parse_query/parse_query.js#L17-L19), but I removed this call because it seemed unnecessary after the updates I made. Can you think of any reason we'd need it?